indico/modules/events/abstracts/controllers/email_templates.py

```diff
@@ -0,0 +1,123 @@
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request
+
+from indico.core.db import db
+from indico.modules.events.abstracts.controllers.management import RHManageAbstractsBase
+from indico.modules.events.abstracts.forms import (EmailTemplateRuleFormEdit, EmailTemplateTextFormEdit,
+                                                   EmailTemplateFormCreate)
+from indico.modules.events.abstracts.models.email_templates import AbstractEmailTemplate
+from indico.modules.events.abstracts.util import build_default_email_template, create_mock_abstract
+from indico.util.placeholders import replace_placeholders
+from indico.web.flask.templating import get_template_module
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data, jsonify_template
+
+
+class RHEmailTemplateList(RHManageAbstractsBase):
+    """Display list of e-mail templates."""
+
+    def _process(self):
+        return jsonify_template('events/abstracts/management/notification_tpl_list.html', event=self.event_new,
+                                email_tpls=self.event_new.abstract_email_templates)
+
+
+class RHEmailTemplateAdd(RHManageAbstractsBase):
+    """Add a new e-mail template."""
+
+    def _process(self):
+        form = EmailTemplateFormCreate(obj=FormDefaults(), event=self.event_new)
+        if form.validate_on_submit():
+            new_tpl = build_default_email_template(self.event_new, form.default_tpl.data)
+            form.populate_obj(new_tpl)
+            self.event_new.abstract_email_templates.append(new_tpl)
+
+            tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
+            return jsonify_data(html=tpl.render_notification_list(
+                self.event_new, self.event_new.abstract_email_templates))
+        else:
+            return jsonify_template('events/abstracts/management/notification_tpl_form.html', form=form)
+
+
+class RHEMailTemplateModifBase(RHManageAbstractsBase):
+    """Base RH for a specific subcontribution"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.email_tpl
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageAbstractsBase._checkParams(self, params)
+        self.email_tpl = AbstractEmailTemplate.get_one(request.view_args['email_tpl_id'])
+
+
+class RHEmailTemplateRuleEdit(RHEMailTemplateModifBase):
+    """Add a new e-mail template."""
+
+    def _process(self):
+        form = EmailTemplateRuleFormEdit(obj=FormDefaults(self.email_tpl), event=self.event_new)
```

Comment:
`obj=self.email_tpl` should work fine here - FormDefaults is just needed if you need more than simple attribute access (e.g. whitelist/blacklist attrs or specify custom values)